hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,171 @@
+For hardware generated commands, there are two modes that allow this.
+One is the auto request mode, where two FIFOs are provided to send commands and additional data. 
+The general operation of this mode is that the CSRNG instance is set up by firmware, then the FIFOS are preloaded with commands.
+One FIFO can be programmed to send `generate` commands.
+The other FIFO can be programmed to send `reseed` commands.
+There is a configurable register that sets the number of requests between reseeds.
+This register determines how many `generate` commands are sent before a `reseed` command is sent.
+The process of sending these two commands will repeat forever until the module is reset or the enable bit is cleared.
+
+## Example Topology
+The diagram below shows an example topology where two EDN modules are used to distribute CSRNG `genbits` to peripheral modules.
+
+![EDN Example Topology Diagram](edn_top_diag.svg)
+
+
+## Compatibility
+Commands generated toward the CSRNG application interface must conform to the CSRNG application interface specification.
+
+
+# Theory of Operations
+
+The first step that must be done to enable the EDN block is to configure the  {{< regref "CTRL" >}} register.
+This will enable the block in general, and then determine which of three modes are to be enabled.
+Broadcast mode can be selected independent of auto request mode or boot request mode.
+A choice between auto request mode and boot request mode should be made.
+
+Once a CSRNG command request mode has been configured, the EDN block will start requesting to the CSRNG block.
+Data will be returned from the specific CSRNG instance that the EDN block is connected to.
+It is assumed that the EDN block has sent an `instantiate` application command before sending any `generate` application commands.
+
+Any peripheral module that is on a request/ack bus should expect the EDN block to fetch `genbits` only when requested.
+Conversely, any peripheral module that is on a broadcast bus should expect the EDN block to fetch `genbits` continuously without a request.
+
+When in auto request mode, the {{< regref "SUM_STS" >}} register will indicate when this mode is done, as in the state machine is idle.
+For very long requests, this indication will be always on unless the module is disabled.
+
+## Block Diagram
+
+![EDN Block Diagram](edn_blk_diag.svg)
+
+## Hardware Interfaces
+
+{{< hwcfg "hw/ip/edn/data/edn.hjson" >}}
+
+## Design Details
+
+### EDN Initialization
+
+After power-up, the EDN block is disabled.
+A single TLUL configuration write to the  {{< regref "CTRL" >}} register will start the boot mode process.
+CSRNG application commands (parameterized) will be sent immediately.
+Once these commands have completed, a status bit will be set.
+At this point, firmware can later come and reconfigure the EDN block for a different mode of operation.
+
```

Comment:
could you add some description here (the ROM people will want it) on the enable order (or lack thereof) between this module and CSRNG / ES?

It would be quite awesome if software could just blindly enable all 3 at boot and just wait for everything to finish (and based on what I've seen, i think this is the case also?)

Reply:
Sequence added. It is quite awesome, and that is how it works - three config writes to each of the entropy system blocks. Shake, bake, and only 2 mSec  later entropy is served to the OTP block on the broadcast bus.

Reply:
hmm... well now my earlier comment about the broadcast bus use case might be sort of explained here, but I still am left wondering if it's ok for us to potentially consume the same entropy in multiple places.  (It might be, but I'm not sure yet.)

Reply:
hey chris, yeah i agree that re-using isn't ideal.. but I was hoping to restrict this group to non-masking usage.
So this i was thinking would be more for blocks like alert_handler and flash that is not trying to do anything cryptographically secure, but just using to create some random-ness. 

In an email with some time ago, I suggested my thinking as follows:  And only the last group should be using this broadcast.
This of course also assumes that if these blocks are lfsr based, their default seeds need to be unique.

masking usage - aes / kmac / obtn 
key / mask / clock jitter usage - key manager / otp / ast
any randomness usage - all the other things I mentioned (alert handler / life cycle / flash).

Reply:
Ah, I remember this email, I just hadn't mapped it onto the EDN broadcast facility.

If the block is using the entropy as a nonce/mask, it's definitely not going to be OK.  For alert handler / flash it's probably going to be OK, but this should be flagged as something to review.  (Inclusion in the broadcast group is a compile/build-time decision right?  Not something dynamic? *shudder*)

Reply:
definitely not dynamic.
And yes totally agree that we should review these and make sure we agree it's the right hook-up.

Shall we add it to the design guidance doc? :)
We should also definitely review all the default seeds for LFSR

Reply:
Nope it is certainly not dynamic.